### PR TITLE
fix(#33) NoClassDefFoundError: javax.transaction.UserTransaction on W…

### PR DESCRIPTION
**Pull request by rimuln**
…ildfly 28 /Jakarta EE 10

#### Short description of what this resolves:
Fix for #33 

#### Changes proposed in this pull request:

- Increased jboss-parent to 39 (Same as Arquillian Core)
- Increased compiler to 1.8 as maven 3.9.0 doesn't support current.
- Added to jta module Eclipse Transformer plugin that creates Jakarta 10 related library with classifier jakarta
- Updated bom to managed this new jar with classifier too.

### CODE REVIEW

**Comment by rimuln** [comment]
@chengfang 
Hi, is some easy way how split it? Or only new branch, cherry pick etc.?

From long term I agree that direct fix source code for Jakarta EE is better. But we need it working in next few days during our 14days sprint. If will not be release of this and fixed Arquilian BOM for 1.7.0.Final (https://github.com/arquillian/arquillian-core/issues/475) we will do own internal company releases for some time.

It is in fact 1 class in this module that contains import javax.transaction.UserTransaction, if I looked correctly.

Current solution have benefit that still library supports Java EE and Jakarta EE. It is same as I found in QueryDSL library.

**Comment by chengfang** [comment]
@rimuln I think the easiest way is to include the relevant changes in a separate branch of yours, and submit a new PR.

File: pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
   </licenses>
 
   <properties>
-    <version.arquillian_core>1.1.13.Final</version.arquillian_core>
+    <version.arquillian_core>1.7.1.Final-SNAPSHOT</version.arquillian_core>
```
Comment (chengfang) [inline pom.xml:29]: We should not depend on a snapshot version.  

To depend on arquillian-core 1.7.1.Final, we'll need to wait for its release.

File: pom.xml [hunk 2]
```diff
@@ -35,9 +35,9 @@
     <version.org.jboss.jbossas>7.1.1.Final</version.org.jboss.jbossas>
 
     <!-- override from parent -->
-    <maven.compiler.target>1.5</maven.compiler.target>
-    <maven.compiler.source>1.5</maven.compiler.source>
-    <maven.min.version>3.0.3</maven.min.version>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.min.version>3.8.1</maven.min.version>
```
Comment (chengfang) [inline pom.xml:40]: Once jboss-parent is upgraded to 39, these override should no longer be needed.

File: pom.xml (hunk 1 above)
Comment (rimuln) [reply pom.xml:29]: Problem is wrong deployment of BOM see (https://github.com/arquillian/arquillian-core/issues/475). Not sure who and when will fix it and how:(

File: pom.xml (hunk 2 above)
Comment (rimuln) [reply pom.xml:40]: ok I will remove it

**Comment by rimuln** [comment]
ok I will another branch and PR for the main pom.xml later evening

File: pom.xml (hunk 1 above)
Comment (chengfang) [reply pom.xml:29]: looks like arquillian-bom was left out when releasing 1.7.0.Final. I'll take a look and see if we can release `org.jboss.arquillian:arquillian-bom` 1.7.0.Final.

Comment (chengfang) [reply pom.xml:29]: `org.jboss.arquillian:arquillian-bom` has just been released and will be sync'ed to central shortly:
https://repository.jboss.org/nexus/content/groups/public/org/jboss/arquillian/arquillian-bom/1.7.0.Final/

**Comment by rimuln** [comment]
@chengfang so what next steps? I see that you merge jboss-parent to master create another PR for 1.7.0.Final as it seems that now it is correctly synced? 
What about the jakarta create direct replace in JTaTransactionProvider class to change from import javax.transaction to jakarta.transaction? or first 1.0.6 will be last for both with transformer and next 2.0.0  or just 1.1.0 only for jakarta e.g.?

**Comment by chengfang** [comment]
As outlined in https://github.com/arquillian/arquillian-extension-transaction/discussions/42 , the current plan is to do a direct pkg replacement in source code in the master branch, and when everything is ready the next release 2.x will include the support for jakarata.* api. No transformer is needed.

The future branch to be created 1.x will be for javax.* api only.

@rimuln Would you mind working on the jakarta.* api changes?

**Comment by rimuln** [comment]
I've created PR with the change to Jakarta. https://github.com/arquillian/arquillian-extension-transaction/pull/45

**Pull request closed without merging**